File: 백준/C++/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
using namespace std;

int main(void) {
	
    ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
    
	int n, num;
	cin >> n;
	stack <int> s;
	string input;

	for (int i = 0; i < n; i++) {
		cin >> input;
		if (input == "push") {
			cin >> num;
			s.push(num);
		}
		else if (input == "pop") {
			if (!s.empty()) {
				cout << s.top() << "\n";
				s.pop();
			}
			else {
				cout << "-1" <<"\n";
			}

		}
		else if (input == "size") {
			cout << s.size() << "\n";
		}
		else if (input == "empty") {
			if (!s.empty()) cout << "0" <<"\n";
			else cout << "1" << "\n";
		}
		else if (input == "top") {
			if (!s.empty()) cout << s.top() << "\n";
			else cout << "-1" << "\n";
		}
	}

	return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
                
        int testCase = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();
        
        while(testCase>0){
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            if(command.equals("push")){
                int value = Integer.parseInt(st.nextToken());
                stack.push(value);
            }else if(command.equals("top")){
                if(stack.size() ==0){
                    sb.append(-1).append("\n");
                }else{
                    sb.append(stack.peek()).append("\n");
                }               
            }else if(command.equals("size")){
                sb.append(stack.size()).append("\n");
            }else if(command.equals("empty")){
                if(stack.isEmpty()){
                    sb.append(1).append("\n");
                }else{
                    sb.append(0).append("\n");
                }                  
            }else if(command.equals("pop")){
                if(stack.size() ==0){
                    sb.append(-1).append("\n");
                }else{                    
                    sb.append(stack.pop()).append("\n");
                }  
            }
            testCase--;
        }
        System.out.print(sb);
    }
}